scapy/contrib/tzsp.py

```diff
@@ -0,0 +1,474 @@
+#! /usr/bin/env python
+#
+# scapy.contrib.description = TZSP
+# scapy.contrib.status = loads
+
+"""
+    TZSP - TaZmen Sniffer Protocol
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    :author:    Thomas Tannhaeuser, hecke@naberius.de
+    :license:   GPLv2
+
+        This module is free software; you can redistribute it and/or
+        modify it under the terms of the GNU General Public License
+        as published by the Free Software Foundation; either version 2
+        of the License, or (at your option) any later version.
+
+        This module is distributed in the hope that it will be useful,
+        but WITHOUT ANY WARRANTY; without even the implied warranty of
+        MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+        GNU General Public License for more details.
+
+    :description:
+
+        This module provides Scapy layers for the TZSP protocol.
+
+        references:
+            - https://en.wikipedia.org/wiki/TZSP
+            - https://web.archive.org/web/20050404125022/http://www.networkchemistry.com/support/appnotes/an001_tzsp.html
+
+    :NOTES:
+        - to allow Scapy to dissect this layer automatically, you need to bind the TZSP layer to UDP using
+          the default TZSP port (0x9090), e.g.
+
+            bind_layers(UDP, TZSP, sport=TZSP_PORT_DEFAULT)
+            bind_layers(UDP, TZSP, dport=TZSP_PORT_DEFAULT)
+
+        - packet format definition from www.networkchemistry.com is different from the one given by wikipedia
+        - seems Wireshark implements the wikipedia protocol version (didn't dive into their code)
+        - observed (miss)behavior of Wireshark (2.2.6)
+          - fails to decode RSSI & SNR using short values - only one byte taken
+          - SNR is labeled as silence
+          - WlanRadioHdrSerial is labeled as Sensor MAC
+          - doesn't know the packet count tag (40 / 0x28)
+
+"""
+from scapy.contrib.avs import AVSWLANHeader
+from scapy.error import warning
+from scapy.fields import ByteField, ShortEnumField, IntField, FieldLenField
+from scapy.layers.dot11 import Packet, Dot11, PrismHeader
+from scapy.layers.l2 import Ether, \
+    struct, StrLenField, ByteEnumField, \
+    ShortField, Scapy_Exception
```

Comment:
Please import things from their correct file:
```
Scapy_Exception                        --> scapy.error
StrLenField, ByteEnumField, ShortField --> scapy.fields
struct                                 --> struct
```

Reply:
thanks for the review. seems pycharm took a shortcut when adding those imports automatically - but at the end of the day its my duty...